doc/manual/metaprogramming.rst

```diff
@@ -134,19 +162,88 @@ ambiguity in parsing.:
     julia> :(::)
     :(::)
 
-:obj:`Symbol`\ s can also be created using :func:`symbol`, which takes
-a character or string as its argument:
+Expressions and evaluation
+--------------------------
+
+Quoting
+~~~~~~~
+
+There is special syntax to create expression objects without using the
+explicit :obj:`Expr` constructor above. A short form for inline expressions
+uses the ``:`` followed by a single parenthesized expression. Here is an
+example of the short form used to quote an arithmetic expression:
 
 .. doctest::
 
-    julia> symbol('\'')
-    :'
+    julia> ex = :(a+b*c+1)
+    :(a + b * c + 1)
 
-    julia> symbol("'")
-    :'
+    julia> typeof(ex)
+    Expr
+
+(to view the structure of this expression, try ``ex.head`` and ``ex.args``,
+or use :func:`dump` as above).
 
-:func:`eval` and Interpolation
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Note that equivalent expressions may be constructed using :func:`parse` or
+the direct :obj:`Expr` form:
+
+.. doctest::
+
+    julia>      :(a + b*c + 1)  ==
+	   parse("a + b*c + 1") ==
+	   Expr(:call, :+, :a, Expr(:call, :*, :b, :c), 1)
+    true
+
+Expressions provided by the parser generally only have symbols, other
+expressions, and literal values as their args, whereas expressions
+constructed by Julia code can have arbitrary run-time values
+without literal forms as args. In this specific example, ``+`` and ``a``
+are symbols, ``*(b,c)`` is a subexpression, and ``1`` is a literal
+64-bit signed integer.
+
+There is a second form of quoting for multiple expressions:
+blocks of code enclosed in ``quote ... end``
+
+.. doctest::
+
+    julia> ex = quote
+		    x = 1
+		    y = 2
+		    x + y
+		end
+    quote  # none, line 2:
+	x = 1 # line 3:
+	y = 2 # line 4:
+	x + y
+    end
+
+    julia> typeof(ex)
```

Comment:
```
julia> Meta.show_sexpr(ex)
(:block,
  (:line, 2, :none),
  (:(=), :x, 1),
  :( # line 3:),
  (:(=), :y, 2),
  :( # line 4:),
  (:call, :+, :x, :y)
)
```

might also be interesting here, to show that `quote ... end` simply wraps the quoted expressions in a block (expression).